velox/exec/tests/HashJoinTest.cpp

```diff
@@ -2344,10 +2344,84 @@ TEST_P(MultiThreadedHashJoinTest, leftJoin) {
       .buildVectors(std::move(buildVectors))
       .buildProjections({"c0 AS u_c0", "c1 AS u_c1"})
       .joinType(core::JoinType::kLeft)
-      //.joinOutputLayout({"row_number", "c0", "c1", "u_c1"})
       .joinOutputLayout({"row_number", "c0", "c1", "u_c0"})
       .referenceQuery(
           "SELECT t.row_number, t.c0, t.c1, u.c0 FROM t LEFT JOIN u ON t.c0 = u.c0")
+      .verifier([&](const std::shared_ptr<Task>& task, bool /*unused*/) {
+        int nullJoinBuildKeyCount = 0;
+        int nullJoinProbeKeyCount = 0;
+
+        for (auto& pipeline : task->taskStats().pipelineStats) {
```

Comment:
perhaps, extract this logic into a helper function to avoid copy-paste

Reply:
You mean the looping? Its a huge test with 2-3 loops like this - so not very enticing. Unless you mean something else

Reply:
The logic for extracting nullJoinBuildKeyCount and nullJoinProbeKeyCount from task stats repeats twice. Can it be extracted into a helper function?

auto [numBuildNulls, numProbeNulls] = extractNumNulls(task->taskStats);